frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.tsx

```diff
@@ -51,25 +56,27 @@ function FilterValuePicker({
     );
   }
 
-  if (data.length > 0 && (data.length <= MAX_INLINE_OPTIONS || !compact)) {
+  if (fieldData && canListFieldValues(fieldData, isCompact)) {
     return (
       <ListValuePicker
-        data={data}
-        value={value}
+        fieldValues={fieldData.values}
+        selectedValues={selectedValues}
         placeholder={t`Search the list`}
-        compact={compact}
+        isCompact={isCompact}
         onChange={onChange}
       />
     );
   }
 
-  if (fieldId != null && hasFieldValues === "search") {
+  if (canSearchFieldValues(fieldInfo)) {
     const columnInfo = Lib.displayInfo(query, stageIndex, column);
 
     return (
       <SearchValuePicker
-        fieldId={fieldId}
-        value={value}
+        fieldId={checkNotNull(fieldInfo.fieldId)}
+        searchFieldId={checkNotNull(fieldInfo.searchFieldId)}
+        fieldValues={fieldData?.values ?? []}
```

Comment:
It can be non-empty when `has_field_values` is `list` but `has_more_values` is `true`. Otherwise we don't load the field values and the whole object can be `undefined`.